src/logging/Logger.cpp

```diff
@@ -1,53 +1,158 @@
-#include "Logger.hpp"
 #include <boost/log/attributes/constant.hpp>
-#include <boost/log/attributes/mutable_constant.hpp>
+#include <boost/log/attributes/function.hpp>
 #include <boost/log/attributes/named_scope.hpp>
-#include <boost/log/core.hpp>
-#include <boost/log/expressions.hpp>
+#include <boost/log/attributes/timer.hpp>
+#include <boost/log/sources/severity_feature.hpp>
 #include <boost/log/trivial.hpp>
 #include <boost/log/utility/setup/common_attributes.hpp>
-#include <iosfwd>
 #include <utility>
+#include <utils/assertion.hpp>
+
+#include "logging/LogConfiguration.hpp"
+#include "logging/Logger.hpp"
 
 namespace precice {
 namespace logging {
 
+/// A feature that adds \ref LogLocation info to log attributes
+template <class BaseT>
+class precice_feature : public BaseT {
```

Comment:
Any reason to deviate from our naming convention? `PreciceFeature` ?

Reply:
It forces me to butcher our project name, which hurts my heart.
Alternatives are welcome :smiley: